### PR TITLE
doc: added DeviceBufferAttr and DeviceDebugAttr in Sphinx.

### PR DESCRIPTION
**Pull request by cristi-iacob**
Added the Sphinx documentation for the DeviceBufferAttr and DeviceDebugAttr classes under the Device menu, as stated here #449.

Signed-off-by: Cristi Iacob <cristian.iacob@analog.com>

### CODE REVIEW

**Comment by rgetz** [comment]
You added more than that thought - didn't you? 

For example - you added buffer set_blocking_mode, but that doesn't come up in the doc yet.

You added the the enums - they don't show up in the doc? (I'm not a sphinx/python expert to know that that is standard practice or not).

**Comment by tfcollins** [comment]
Sorry I missed some stuff in the original issue.

Need the following too:
- ChannelModifier
- DataFormat
- ChannelType
- Channel.convert
- Buffer.cancel
- Buffer.get_data
- Buffer.set_blocking_mode
- Buffer.poll_fd
- Buffer.step
- Device.context

I think that should be all the public properties and classes.

**Comment by cristi-iacob** [comment]
I have added the missing documentation.

**Comment by rgetz** [comment]
I don't find any Buffer.get_data - the rest is there.

**Comment by rgetz** [comment]
and thanks for update.

**Comment by cristi-iacob** [comment]
I have discussed with @tfcollins and decided to not implement buffer.get_data and buffer.set_data (and their analogues for device and channel) because these methods are used only for internal calls and therefore is not necessary to implement bindings for them. 
I just forgot to delete it from iio.py in the first place, but I have removed it in this PR.

**Comment by rgetz** [comment]
Ok - sounds great thanks.

**Comment by rgetz** [comment]
What about `iio_channel_convert_inverse` Convert the sample from host format to hardware format? Isn't that needed for some DACs?

**Comment by cristi-iacob** [comment]
I have now implemented iio_convert_inverse method as well.
Also I found a bug in the Channel.convert() method and solved it.

**Comment by rgetz** [comment]
I think there is some weirdness in here? It looks like there are identical commits in the PR?

File: bindings/python/iio.py [hunk 1]
```diff
@@ -123,7 +123,7 @@ class ChannelModifier(Enum):
 	IIO_MOD_H2 = 43
 
 class ChannelType(Enum):
-	"""Contains the type of an IIO channel."""
+	"""Contains the types of an IIO channel."""
```
[the remark this answers is not in the document]
Comment (cristi-iacob) [reply bindings/python/iio.py]: Solved.

**Comment by cristi-iacob** [comment]
I have solved the problem with the commits, I think I have made a wrong rebase before, sorry for that.
I have also corrected the type-issue from the documentation of the ChannelType and ChannelModifier classes.

**Comment by rgetz** [comment]
@tfcollins can you also have a look?

**Pull request merged**